crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -821,12 +816,18 @@ impl AuthorityPerEpochStore {
             ?assigned_versions,
             "set_assigned_shared_object_versions"
         );
-        self.get_or_init_next_object_versions(
-            certificate.data().transaction_data(),
-            assigned_versions.iter().map(|(id, _)| *id),
-            parent_sync_store,
-        )
-        .await?;
+
+        #[allow(clippy::needless_collect)]
```

Comment:
nit: It does seem like needless collect, do we actually need it?

Reply:
without we get a horrible borrow checker error that i couldn't see any way to fix